arm/Microsoft.ContainerService/managedClusters/agentPools/deploy.bicep

```diff
@@ -5,17 +5,503 @@ param managedClusterName string
+
+@description('Optional. Subnet ID for the pod IPs. If omitted, pod IPs are statically assigned on the node subnet (see vnetSubnetID for more details). This is of the form: /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/virtualNetworks/{virtualNetworkName}/subnets/{subnetName}	')
+param podSubnetId string = ''
+
+@description('Optional. The ID for the Proximity Placement Group.')
+param proximityPlacementGroupId string = ''
+
+@description('Optional. Describes how VMs are added to or removed from Agent Pools. See billing states (https://docs.microsoft.com/en-us/azure/virtual-machines/states-billing).')
+@allowed([
+  'Deallocate'
+  'Delete'
+])
+param scaleDownMode string = 'Delete'
+
+@description('Optional. The eviction policy specifies what to do with the VM when it is evicted. The default is Delete. For more information about eviction see spot VMs	')
+@allowed([
+  'Deallocate'
+  'Delete'
+])
+param scaleSetEvictionPolicy string = 'Delete'
+
+@description('Optional. The Virtual Machine Scale Set priority.')
+@allowed([
+  'Regular'
+  'Spot'
+  ''
+])
+param scaleSetPriority string = ''
+
+@description('Optional. Possible values are any decimal value greater than zero or -1 which indicates the willingness to pay any on-demand price. For more details on spot pricing, see spot VMs pricing (https://docs.microsoft.com/en-us/azure/virtual-machines/spot-vms#pricing)')
+param spotMaxPrice int = -1
+
+@description('Optional. Tags of the resource.')
+param tags object = {}
+
+@description('Optional. The type of Agent Pool.')
+param type string = ''
+
+@description('Optional. This can either be set to an integer (e.g. "5") or a percentage (e.g. "50%"). If a percentage is specified, it is the percentage of the total agent pool size at the time of the upgrade. For percentages, fractional nodes are rounded up. If not specified, the default is 1. For more information, including best practices, see: /azure/aks/upgrade-cluster#customize-node-surge-upgrade')
+param maxSurge string = ''
+
+@description('Optional. VM size. VM size availability varies by region. If a node contains insufficient compute resources (memory, cpu, etc) pods might fail to run correctly. For more details on restricted VM sizes, see: /azure/aks/quotas-skus-regions')
+param vmSize string = 'Standard_D2s_v3'
+
+@description('Optional. Node Subnet IDIf this is not specified, a VNET and subnet will be generated and used. If no podSubnetID is specified, this applies to nodes and pods, otherwise it applies to just nodes. This is of the form: /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/virtualNetworks/{virtualNetworkName}/subnets/{subnetName}	')
+param vnetSubnetId string = ''
+
+@description('Optional. Determines the type of workload a node can run.')
+param workloadRuntime string = ''
 
 resource managedCluster 'Microsoft.ContainerService/managedClusters@2021-08-01' existing = {
```

Comment:
I'd suggest to move this reference down to the agent pool resource

Reply:
Moved reference down.